src/os/file.go

```diff
@@ -115,10 +116,23 @@ func (f *File) Read(b []byte) (n int, err error) {
 // It returns the number of bytes read and any error encountered, possible io.EOF.
 // At end of file, Read returns 0, io.EOF.
 func (f *File) ReadAt(b []byte, offset int64) (n int, err error) {
-	n, err = f.handle.ReadAt(b, offset)
-	if err != nil && err != io.EOF {
-		err = &PathError{"readat", f.name, err}
+	if offset < 0 {
+		return 0, &PathError{Op: "readat", Path: f.name, Err: errors.New("negative offset")}
```

Comment:
Note that `errors.New` causes a heap allocation. You can avoid this by declaring it as an (unexported) global variable.
(Just a suggestion, not critical since it's not for a resource constrained system).